dwave/preprocessing/include/dwave/presolve.h

```diff
@@ -285,6 +285,46 @@ void Presolver<bias_type, index_type, assignment_type>::apply() {
             changes = remove_zero_biases(model_.constraint_ref(c)) || changes;
         }
 
+        // *-- clear out small linear biases in the constraints
+        // todo: temporarily hard-coded the FEASIBILITY_TOLERANCE here.
+        // todo: ideally this should be added to dimod.
+        bias_type FEASIBILITY_TOLERANCE = 1.0e-6;
+        for (size_type c = 0; c < model_.num_constraints(); ++c) {
+            auto& constraint = model_.constraint_ref(c);
+            std::vector<index_type> small_biases;
+            std::vector<index_type> small_biases_temp;
+            bias_type reduction = 0;
+            bias_type reduction_limit = 0;
+
+            for (auto& v : constraint.variables()) {
+                // ax ◯ c
+                bias_type a = constraint.linear(v);
+                bias_type lb = constraint.lower_bound(v);
+                bias_type ub = constraint.upper_bound(v);
+                bias_type v_range = ub - lb;
+
+                if (abs(a) < 1.0e-3 && abs(a) * v_range * constraint.num_variables()
+                < 1.0e-2 * FEASIBILITY_TOLERANCE) {
+                small_biases_temp.emplace_back(v);
+                reduction += a * lb;
+                reduction_limit += abs(a) * v_range;
+                }
+
+                if (abs(a) < 1.0e-10)  small_biases.emplace_back(v);
+                continue;
+            }
+
+            if (reduction_limit < 1.0e-1 * FEASIBILITY_TOLERANCE) {
+                constraint.set_rhs(constraint.rhs() - reduction);
+                for (auto& u : small_biases_temp) small_biases.emplace_back(u);
+            }
+
+
+            for (auto& v : small_biases) {
+                constraint.remove_variable(v);
```

Comment:
This looks like it might fail when variable is added twice to `small_biases`, and that could happen when range is huge (and bias very small) and you add the same variable to `small_biases_temp` for conditional removal and to `small_biases` for unconditional removal.

Reply:
I checked this and looks like `remove_variable()` does not fail if you apply it to a variable that does not exist.